index.js

```diff
@@ -1 +1,103 @@
-// Write the helper here :)
+(function(root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define(['dustjs-linkedin'], factory);
+  } else if (typeof exports === 'object') {
+    module.exports = factory(require('dustjs-linkedin'));
+  } else {
+    factory(root.dust);
+  }
+}(this, function(dust) {
+
+  dust.helpers = dust.helpers || {};
+
+  dust.helpers.component = function component(chunk, context, bodies, params) {
+
+    // The component name is required to load and render the template.
+    if (!params.name) {
+      return chunk;
+    }
+
+    bodies = bodies || {};
+    params = params || {};
+
+    // Keep a clear reference to the original context.
+    var originalContext = context;
+
+    // By default, components are completely isolated from the outside template
+    // context. So we start with a new, empty context.
+    var componentContext = dust.makeBase(context.global);
+
+    // Set the template to render.
+    var templateName = componentContext.templateName = params.name;
+
+    // Params to be passed to the component template.
+    var componentParams = [];
+
+    // Some param names are reserved:
+    // - `name` is for choosing the component to render and isn’t passed to
+    //   the template.
+    // - `ctx` is reserved for passing the entire component context in one
+    //   param, which is sometimes convenient. Even though Dust has its own
+    //   {tag:context /} syntax for this, We can’t respect it because in a
+    //   helper there is no way to differentiate explicit from inherited
+    //   context, and we’re avoiding inherited context by design. This param is
+    //   not passed to the template.
+    // - `body` is reserved as the key in which the default dust body is
+    //   provided to the template. It therefore can’t be used as a param name.
+    var reservedParamNames = ['name', 'ctx', 'body'];
+
+    // If and only if an explit context was provided, set up that context.
+    // Otherwise, the context available to the component template is limited
+    // to the params and bodies passed in.
+    if (params.ctx) {
+      componentContext = componentContext.push(params.ctx);
+
+      // The empty context created above is now set as the tail; we can safely
+      // remove it.
+      if (componentContext.stack) {
+        componentContext.stack.tail = undefined;
+      }
```

Comment:
If the componentContext created above can sometime have something in it (if context.global does) then we probably don’t want to do this.